asset/src/partition_by_date/schema.ts

```diff
@@ -0,0 +1,24 @@
+import { ConvictSchema } from '@terascope/job-components';
+import { PartitionByDateConfig } from './interfaces';
+
+export default class Schema extends ConvictSchema<PartitionByDateConfig> {
+    build() {
+        return {
+            field: {
+                doc: 'Which field in each data record contains the date to use for timeseries',
+                default: 'date',
+                format: 'required_String'
+            },
+            resolution: {
+                doc: 'Type of timeseries data',
+                default: 'daily',
+                format: ['daily', 'monthly', 'yearly']
+            },
+            delimiter: {
+                doc: 'Char which will be used to join by',
+                default: '/',
+                format: 'optional_String'
```

Comment:
Using a `/` works well for the file assets but I'd imagine this wouldn't be best default for all use cases. So we should probably make it a `required_String` without a default.